client/gribiclient.go

```diff
@@ -345,7 +347,23 @@ func (c *Client) Connect(ctx context.Context) error {
 				log.V(2).Infof("shutting down recv goroutine")
 				return
 			}
-			if done := respHandler(stream.Recv()); done {
+			resp, err := stream.Recv()
+			trailer := stream.Trailer()
+
+			if len(trailer) != 0 {
```

Comment:
Can you add a comment indicating a link to the grpc-core implementation to explain why this is required? It didn't make sense to me without this.

I suggest something like:

```golang
// grpc-core does not support populating the status details, therefore some
// implementations may support this using trailing metadata. The specific keys
// are defined above as constants 
// (GRIBIErrorTrailingMetadataName and GRIBIErrorTrailingMetadataContents). If
// these keys exist in the metadata then we append these values to the error
// that is returned to the client.
```

Reply:
Thanks for the suggestion
I have implemented it in latest commit